backend/persistent/datastores/DataWorld.js

```diff
@@ -0,0 +1,118 @@
+import fetch from 'node-fetch';
+import url from 'url';
+
+import Logger from '../../logger';
+
+function parseUrl(datasetUrl) {
+  const pathnameArray = url.parse(datasetUrl).pathname.split('/');
+  return {
+    owner: pathnameArray[1],
+    id: pathnameArray[2]
+  };
+}
+
+export function connect(connection) {
+  const { owner, id } = parseUrl(connection.url);
+  return new Promise((resolve, reject) => {
+    fetch(`https://api.data.world/v0/datasets/${owner}/${id}/`, {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${connection.token}`,
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+        'User-Agent': `Falcon/Plotly - ${process.env.npm_package_version}`
+      }
+    })
+    .then(res => res.json())
+    .then(json => {
+      // json.code is defined only when there is an error
+      if (json.code) {
+        reject(json);
+      } else {
+        resolve();
+      }
+    })
+    .catch(err => {
+      Logger.log(err);
+      throw new Error(err);
+    });
+  });
+ }
+
+export function tables(connection) {
+  return new Promise((resolve, reject) => {
+    query('SELECT * FROM Tables', connection).then((res) => {
+      const allTables = res.rows.map((table) => {
+        return table[0];
+      });
+
+      resolve(allTables);
+    })
+    .catch(err => {
+      reject(err);
+    });
+  });
+}
+
+export function schemas(connection) {
+  return new Promise((resolve) => {
+    query('SELECT * FROM TableColumns', connection).then((res) => {
+      const rows = res.rows.map((table) => {
+        const tableName = table[0];
+        const columnName = table[3];
+        // Extract the datatype from datatype url e.g. http://www.w3.org/2001/XMLSchema#integer
+        const columnDataType = /#(.*)/.exec(table[6])[1];
+
+        return [
+          tableName,
+          columnName,
+          columnDataType
+        ];
+      });
+
+      resolve({
+        columnNames: [ 'tablename', 'column_name', 'data_type' ],
+        rows
+      });
+    })
+    .catch(err => {
+      Logger.log(err);
+      throw new Error(err);
+    });
+  });
+}
+
+export function query(queryString, connection) {
+  const { owner, id } = parseUrl(connection.url);
+  return new Promise((resolve, reject) => {
+    const queryStatement = `${queryString.replace(/-/g, '_')}`;
```

Comment:
This substitution could potentially break the SQL query,

I'd move this code (that replaces `-` with `_`) into `tables` and `schemas` to ensure the user only sees table names without `-`.